adapters/handlers/grpc/v1/prepare_aggregate_reply.go

```diff
@@ -15,13 +15,29 @@ import (
 	"fmt"
 
 	"github.com/weaviate/weaviate/entities/aggregation"
+	"github.com/weaviate/weaviate/entities/models"
+	"github.com/weaviate/weaviate/entities/schema"
 	pb "github.com/weaviate/weaviate/grpc/generated/protocol/v1"
 )
 
-type AggregateReplier struct{}
+type AggregateReplier struct {
+	authorizedGetDataTypeOfProp func(string) (string, error)
+}
 
-func NewAggregateReplier() *AggregateReplier {
-	return &AggregateReplier{}
+func NewAggregateReplier(authorizedGetClass func(string) (*models.Class, error), params *aggregation.Params) *AggregateReplier {
+	return &AggregateReplier{
+		authorizedGetDataTypeOfProp: func(propName string) (string, error) {
+			class, err := authorizedGetClass(string(params.ClassName))
+			if err != nil {
+				return "", fmt.Errorf("get class: %w", err)
+			}
+			schemaProp, err := schema.GetPropertyByName(class, propName)
+			if err != nil {
+				return "", fmt.Errorf("get property by name: %w", err)
+			}
+			return string(schema.DataType(schemaProp.DataType[0])), nil
+		},
```

Comment:
I'm making a note here of how this should be refactored in the future :D

Reply:
yeah :) I have it on my list :)